### PR TITLE
Add support for C++11 override keyword

### PR DESCRIPTION
**Pull request by jwpeterson**
The `override` [keyword](http://en.cppreference.com/w/cpp/language/override) can be used to mark functions that, as the name suggests, override virtual functions from a base class.  Thus not all virtual functions are overrides, but all override functions must be virtual.

This keyword imposes no additional runtime overhead, but provides compile-time checks to ensure that you are actually overriding a virtual and haven't accidentally defined a new function. They also provide extra documentation to the reader of a class: they can now distinguish whether a given virtual function in class X overrides a function from a base of X or is defining a new virtual interface for X and its subclasses.

Refs #655.

### CODE REVIEW

File: include/fe/fe.h [hunk 1]
```diff
@@ -686,7 +690,9 @@ class FEHierarchic : public FE<Dim,HIERARCHIC>
    * to be used in dimension \p Dim.
    */
   explicit
-  FEHierarchic(const FEType& fet);
+  FEHierarchic(const FEType& fet) :
+    FE<Dim,HIERARCHIC> (fet)
+  {}
```
Comment (roystgnr) [inline include/fe/fe.h]: How'd you end up catching these?

Comment (roystgnr) [inline include/fe/fe.h]: Wait, never mind, that's not a catch, just a reorganization.  At some point we should nail down style guidelines about when inline functions should go inside the class definition and when outside.

Comment (jwpeterson) [inline include/fe/fe.h]: Yeah, these changes should be in a separate patch, so they can be reverted if you want, but I highly prefer this version.  It's easier to read and about 75% less lines for one-liner constructors.

Comment (roystgnr) [inline include/fe/fe.h]: As long as they're in a separate commit, they're fine mixed in this PR.

Comment (jwpeterson) [inline include/fe/fe.h]: > As long as they're in a separate commit, they're fine mixed in this PR.

Yeah, they're in 9274862

**Comment by roystgnr** [comment]
I made it up to "Sorry, we could not display the changes to this file because there were too many other changes to display."  You broke GitHub!

No problems in what I saw, though.  If the whole rest of the diff is just more add-libmesh_override-everywhere then go ahead and merge.

**Comment by jwpeterson** [comment]
> No problems in what I saw, though. If the whole rest of the diff is just more add-libmesh_override-everywhere then go ahead and merge.

Thanks for the review.
